pkg/codegen/schema.go

```diff
@@ -53,10 +60,24 @@ func (s Schema) GetAdditionalTypeDefs() []TypeDefinition {
 }
 
 type Property struct {
-	Description   string
-	JsonFieldName string
-	Schema        Schema
-	Required      bool
+	Description        string
+	Required           bool
+	FieldNullable      bool
```

Comment:
This code is a little bit of an ill fit with the rest of the rather complicated code in here.

Required and FieldNullable are kind of related. FieldTag, JsonFieldName, and JsonFieldOmitEmpty are also a bit special case relative to the very general usage of this Property.

What would you think about parsing all x-go-XXX extensions into a map, which you then store on this object, so that it's easier to add future extensions. Then, you create getter functions on Property which look into that map.

For the case of not nullable fields, you could rename Required to required and add a function called Required() which would then be used by the templates, and it will achieve what you want, by using the nullable extension to actually make things required.